core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java

```diff
@@ -81,6 +82,8 @@ public BlockletDataMapIndexWrapper get(TableBlockIndexUniqueIdentifierWrapper id
     List<BlockDataMap> dataMaps = new ArrayList<>();
     if (blockletDataMapIndexWrapper == null) {
       try {
+        boolean isCarbonTableUpdated =
```

Comment:
Why not list files inside segment path once to list al carbodata files and check the existance in that list?

Reply:
Listing files from metadata folder will be less costly as compared to listing files from segment path. Metadata contains only few files but segment path can contain lot of files which will do more RPC calls to namenode
Also listing from segment file path will be very costly in case of flat folder structure OR when storage file system is S3

Reply:
Already list files done to get the locations of files, so better check in the map `carbonDataFileBlockMetaInfoMapping`

Reply:
yes I have checked...I will modify the code and test the performance and then make the changes in PR